src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -228,86 +246,57 @@ private EventsRetryResult retryEventsOnce(String userId, String deviceId, List<E
   }
 
   private List<Event> getEventListToRetry(List<Event> events, Response response) {
-    List<Event> eventsToRetry = events;
+    List<Event> eventsToRetry = new ArrayList<>();
     List<Event> eventsToDrop = new ArrayList<>();
     // Filter invalid event out based on the response code.
     if (response.status == Status.SUCCESS) {
-      return new ArrayList<>();
+        triggerEventCallbacks(events, response.code, "Events sent success.");
+      return eventsToRetry;
+    } else if (response.status == Status.FAILED) {
+        triggerEventCallbacks(events, response.code, "Event sent Failed.");
+        return eventsToRetry;
+    } else if (response.status == Status.UNKNOWN){
+        triggerEventCallbacks(events, response.code, "Unknown response status.");
+        return eventsToRetry;
     } else if (response.status == Status.INVALID) {
-      if ((response.invalidRequestBody != null
-              && response.invalidRequestBody.has("missingField")
-              && response.invalidRequestBody.getString("missingField").length() > 0)
-          || events.size() == 1) {
-        // Return early if there's an issue with the entire payload
-        // or if there's only one event and its invalid
-        triggerEventCallbacks(events, response.code, response.error);
```

Comment:
Support for this case is removed?

Reply:
Still support this case. It's covered in the current code.

Reply:
Added back the special case